src/GafferImage/ImageAlgo.cpp

```diff
@@ -251,35 +283,36 @@ IECore::ConstCompoundDataPtr GafferImage::ImageAlgo::tiles( const ImagePlug *ima
 				// Currently, tiles() is used mostly in tests, so it seem good to verify this.
 				// If tiles() gets used in somewhere performance critical, it would be good to make this check
 				// optional.
-				if( data != ImagePlug::flatTileSampleOffsets() )
+				if( sampleOffsets != ImagePlug::flatTileSampleOffsets() )
 				{
 					throw IECore::Exception( "Accessing tiles on flat image with invalid sample offsets" );
 				}
 			}
+
+			const Gaffer::ThreadState &threadState = Gaffer::ThreadState::current();
+			ImagePlug::ChannelDataScope channelDataScope( threadState );
```

Comment:
This isn't strictly necessary : `channelDataScope( Context::current() )` is sufficient. The only time you need to pass a `ThreadState` is when you are launching tasks yourself. In this case that is all done automatically by ImageAlgo. I think it's worth simplifying this to avoid giving a false impression about when `ThreadState` is needed. The intention in most cases is to provide higher level abstractions like `parallelGatherTiles()` instead of manipulating it manually.